core/typescript/src/lib/bootstrap/pulumi/component.ts

```diff
@@ -0,0 +1,87 @@
+import { ComponentResource, ComponentResourceOptions } from '@pulumi/pulumi';
+
+import {
+  ComponentConfig,
+  GeneralConfig,
+  InfraConfig,
+  NetworkConfig,
+  RelationConfig,
+  ServiceConfig,
+} from '../../config';
+import { Orchestrator } from '../../orchestrator';
+import { InfraPlanConstructs, Planner, ResultHandler } from '../../package';
+import { Custom } from '../../project';
+
+/**
+ * Properties for {@link ProjectComponent}.
+ *
+ * @group For project creators
+ */
+export interface ProjectComponentOpts<
```

Comment:
I feel this should be `Stack` to make the model consistent with the other runtimes. Pulumi does have the concept of a stack though there isn't an a concrete implementation.

<img width="726" alt="image" src="https://user-images.githubusercontent.com/51779/208708257-77cd97e2-0d4a-4e7d-86d7-e91191139b11.png">

<img width="719" alt="image" src="https://user-images.githubusercontent.com/51779/208708635-6a19e2b7-818b-47b7-817f-74ee2601d6eb.png">

Reply:
`Stack` in Pulumi carries a different meaning I believe, and that's why I chose to use `Component` over `Stack`. Same goes for cdk8s where `Chart` is used.

Reply:
makes sense in for cdk8s makes sense.

what's the difference in meaning of `stack` carried by Pulumi? I see there's a difference in the implementation. I might be misunderstanding.

Reply:
https://www.pulumi.com/docs/intro/concepts/stack/ which sounds like env in your language or stage in my language :)

Reply:
Reading a little more about `stack` in CDK and Pulumi worlds. As concepts they sounds identical. Implementation differs a little. A concrete implementation of stack does exist in Pulumi also, it's just doesn't need to be explicitly created/referenced in code unless you want to. 

https://docs.aws.amazon.com/cdk/v2/guide/stacks.html

https://www.pulumi.com/learn/building-with-pulumi/understanding-stacks/

Reply:
Not sure I agree. 

- For AWS CDK, you can have multiple stacks within a single account/env, which is also how cloudformation works (stacks and sub-stacks are CF concepts too).
- For Pulumi, stacks == envs and the example they give seem to confirm that too.

Reply:
OK I think I get what you are saying. 

Pulumi - a project has multiple instances of a stack (same shape different config like dev and prod).

CDK - an `App` can have multiple stacks of different shape. So with CDK you could have a front end stack and backend stack in the same program. can deploy multiple instances for env but this isn't hard enforced as opposed to Pulumi. 

Is that^ how you are seeing it?

I wonder how Pulumi map this with their CDK implementation https://www.pulumi.com/blog/aws-cdk-on-pulumi/

I'll run the example to see what it does.

My main concern here is that we are putting Pulumi component and CDK stack at the same level which is pretty jarring.

Reply:
In CDK `stack` is exclusively used for a unit of work, much like what a component means with Pulumi. It's also used in CloudFormations as a term. I'm guessing CDK stack == Pulumi component is how Pulumi sees it. 
I actually think it's okay if not a true reflection of the underlying IaC runtime varieties. Bear in mind that the code you see here already exists in a folder such as `bootstrap/pulumi` which is supposed to branch out in individual runtimes, therefore it makes more sense to respect and reuse the terms used in them to describe top-level construct accordingly.

Reply:
These are helper code for ease of integration for a specific runtime and visible within a project codebase. The consequence of using a common term here is that people will see this in their project that conforms to a specific runtime which carries its own set of terms. If we want to be term-compliant, we should start from here.

Reply:
Maybe the plural `*Components` will help with the metal model and still map to pulumi concepts.